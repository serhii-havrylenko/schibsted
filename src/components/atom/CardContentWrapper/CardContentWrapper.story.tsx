import { Card } from '@material-ui/core';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import CardContentWrapper from './CardContentWrapper';

storiesOf('CardContentWrapper', module).add('default', () => (
  <Card>
    <CardContentWrapper>
      <span>Vertically aligned card content</span>
      <span>Span 2</span>
    </CardContentWrapper>
  </Card>
));
