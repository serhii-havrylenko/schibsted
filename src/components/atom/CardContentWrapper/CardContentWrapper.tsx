import CardContent from '@material-ui/core/CardContent';
import styled from 'styled-components';

export const CardContentWrapper = styled(CardContent)`
  && {
    display: flex;
    flex-direction: column;
  }
` as typeof CardContent;

export default CardContentWrapper;
