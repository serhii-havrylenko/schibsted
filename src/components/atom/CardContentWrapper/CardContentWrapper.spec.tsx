import { mount } from 'enzyme';
import * as React from 'react';

import CardContentWrapper from './CardContentWrapper';

jest.mock('@material-ui/core/CardContent');

describe('<CardContentWrapper /> tests', () => {
  test('should match snapshot and styles', () => {
    const tree = mount(<CardContentWrapper>Content</CardContentWrapper>);
    expect(tree).toMatchSnapshot();
  });
});
