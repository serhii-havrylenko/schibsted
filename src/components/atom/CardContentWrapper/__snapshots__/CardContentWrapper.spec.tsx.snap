// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContentWrapper /> tests should match snapshot and styles 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<Styled(WithStyles(CardContent))>
  <StyledComponent
    forwardedComponent={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "children": "Content",
              "className": "sc-bdVaJa c0",
            },
            Object {
              "64a55d578f856d258dc345b094a2a2b3": undefined,
              "6fc570d6bd61383819d0f9e7407c452d": undefined,
              "__THEMING__": undefined,
              "d4bd0baacbc52bbd48bbb9eb24344ecd": undefined,
              "muiThemeProviderOptions": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    forwardedRef={null}
  >
    <WithStyles(CardContent)
      className="c0"
    />
  </StyledComponent>
</Styled(WithStyles(CardContent))>
`;
